### ISSUE TITLE
Does not work with async/await

### ISSUE THREAD

**Issue by YoranBrondsema** [san650/ember-cli-page-object#309]
I'm trying to get the addon to work with [ember-native-dom-helpers](https://github.com/cibernox/ember-native-dom-helpers) in component integration tests. When doing something like

````
await page.someAction();
````

it fails at [this line](https://github.com/san650/ember-cli-page-object/blob/master/addon/-private/dsl.js#L15) because the `wait()` helper is not defined as a global in component integration tests. I tried fiddling with the addon and replacing that global `wait` with the `wait` from [ember-test-helpers](https://github.com/emberjs/ember-test-helpers) (as is recommended in the guides https://guides.emberjs.com/v2.13.0/testing/testing-components/#toc_waiting-on-asynchronous-behavior) but that caused a bunch of tests to fail. I would be willing to help out but I think I'll need some help from one of the maintainers of this addon to get this to work.

As a workaround, I explicitly set `wait` as a global in my component integration tests:

````
...
import wait from 'ember-test-helpers/wait';

moduleForComponent('some-component', 'Some component', {
  beforeEach() {
    page.setContext(this);
    window.wait = wait;
  },
  ...
});
````

As you can understand, this is not ideal :-).

**Comment by YoranBrondsema** [comment]
Nevermind, I think I found a solution, see https://github.com/san650/ember-cli-page-object/pull/310.

**Comment by san650** [comment]
@YoranBrondsema thanks for filing the issue. The error is a bit strange, can you post here the page object definition you're using?

Also, just to have a bit more context, it would be great to have a snippet of the test file you used.

**Comment by YoranBrondsema** [comment]
@san650 I'll try to set up a minimum Ember app that reproduces the issue later today. I think that'll be easiest!

**Comment by YoranBrondsema** [comment]
@san650 Here it is: https://github.com/YoranBrondsema/ember-cli-page-object-issue-309. Run `ember s` and in `localhost:4200/tests` the test `Integration | Component | some component` should fail with the same error that I'm getting in my app.

I think the bug occurs because in "normal" non-async/await tests, the `wait` is manually imported (as described in the guide https://guides.emberjs.com/v2.13.0/testing/testing-components/#toc_waiting-on-asynchronous-behavior). However, when moving to async/await, the handling of the Promise is delegated to `ember-cli-page-object` addon. Since we are in a component integration test and not an acceptance, no test helpers (including `wait`) are available in the global scope.

Note that the `ember-native-dom-helpers` addon does something similar to what I propose in https://github.com/san650/ember-cli-page-object/pull/310 (see https://github.com/cibernox/ember-native-dom-helpers/blob/3b79b3cea030a1d9b5451ece6077704928f67e22/addon-test-support/click.js#L29).

I hope this helps!

**Comment by san650** [comment]
Thanks @YoranBrondsema for the detailed info. It really helps me to understand the issue.

I'll try to go over your PR next weekend to ensure it works with all the Ember versions we support.

**Comment by GCorbel** [comment]
I have an issue that seems to be similar. When I have an `async/await` function I have this error : 

    Assertion Failed: You have turned on testing mode, which disabled the run-loop's autorun. You will need to wrap any code with asynchronous side-effects in a run

@YoranBrondsema fix doesn't seems to work.

The issue I have is describe more in detail [here](https://github.com/emberjs/rfcs/issues/175). 

This code doesn't work : 

```
wait(()=> {
  console.log(1);
});
```

I don't have the time to go deeper right now but I will tonight.

**Comment by YoranBrondsema** [comment]
@GCorbel Is that your entire test? And is it an acceptance test or component integration test?

**Comment by san650** [comment]
Fixed by #310
